src/processes.c

```diff
@@ -1376,25 +1280,25 @@ static int ps_read_process (long pid, procstat_t *ps, char *state)
 	}
 
 	/* Convert jiffies to useconds */
-	cpu_user_counter   = cpu_user_counter   * 1000000 / CONFIG_HZ;
-	cpu_system_counter = cpu_system_counter * 1000000 / CONFIG_HZ;
+	cpu_user   = cpu_user   * 1000000 / CONFIG_HZ;
+	cpu_system = cpu_system * 1000000 / CONFIG_HZ;
 	vmem_rss = vmem_rss * pagesize_g;
 
-	ps->cpu_user_counter = cpu_user_counter;
-	ps->cpu_system_counter = cpu_system_counter;
-	ps->vmem_size = (unsigned long) vmem_size;
-	ps->vmem_rss = (unsigned long) vmem_rss;
-	ps->stack_size = (unsigned long) stack_size;
+	ps->counters.cpu_user = cpu_user;
+	ps->counters.cpu_system = cpu_system;
+	ps->gauges.vmem_size = (unsigned long) vmem_size;
+	ps->gauges.vmem_rss = (unsigned long) vmem_rss;
+	ps->gauges.stack_size = (unsigned long) stack_size;
 
 	if ( (ps_read_io (pid, ps)) == NULL)
 	{
 		/* no io data */
-		ps->io_rchar = -1;
-		ps->io_wchar = -1;
-		ps->io_syscr = -1;
-		ps->io_syscw = -1;
-		ps->io_diskr = -1;
-		ps->io_diskw = -1;
+		ps->gauges.io_rchar = -1;
```

Comment:
I wonder if this can be accomplished by simply using `procstat_gauges_init` right after the `memset` in line 1185...

Reply:
In this context, it's ok for vmem_data and vmem_code to be -1 by default, but in other context where the value is simply unsupported it should be zero. I'm afraid of making such a large set of assumptions by removing this conditions where the values are set intentionally.

Reply:
You'll have to keep the `vmem_data` and `vmem_code` assignments, because `procstat_gauges_init` sets them to 0. For the rest of the values, however, this should be fine.